crates/evm/src/tests/fork_tests.rs

```diff
@@ -0,0 +1,807 @@
+    }
+    evm.end_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    evm.finalize_hook(&[99u8; 32].into(), &mut working_set.accessory_state());
+
+    l2_height += 1;
+
+    let receipts: Vec<_> = evm
+        .receipts
+        .iter(&mut working_set.accessory_state())
+        .collect();
+
+    let txs: Vec<_> = evm
+        .transactions
+        .iter(&mut working_set.accessory_state())
+        .collect();
+
+    // Last tx should have failed because cancun is not activated
+    assert!(receipts.last().unwrap().receipt.success);
+    let db_account = DbAccount::new(contract_addr);
+    let storage_value = db_account
+        .storage
+        .get(&U256::ZERO, &mut working_set)
+        .unwrap();
+    assert_eq!(storage_value, U256::from(80));
+}
+
+// tests first part of https://eips.ethereum.org/EIPS/eip-6780
+#[test]
+fn test_self_destruct_restriction() {
+    let contract_balance: u64 = 1000000000000000;
+
+    // address used in selfdestruct
+    let die_to_address = address!("11115497b157177315e1204f52e588b393111111");
+
+    let (config, dev_signer, contract_addr) =
+        get_evm_config(U256::from_str("100000000000000000000").unwrap(), None);
+
+    let (mut evm, mut working_set) = get_evm(&config);
+    let l1_fee_rate = 0;
+    let mut l2_height = 2;
+
+    let soft_confirmation_info = HookSoftConfirmationInfo {
+        l2_height,
+        da_slot_hash: [5u8; 32],
+        da_slot_height: 1,
+        da_slot_txs_commitment: [42u8; 32],
+        pre_state_root: [10u8; 32].to_vec(),
+        current_spec: SovSpecId::Genesis,
+        pub_key: vec![],
+        deposit_data: vec![],
+        l1_fee_rate,
+        timestamp: 0,
+    };
+
+    evm.begin_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    {
+        let sender_address = generate_address::<C>("sender");
+        let context = C::new(sender_address, l2_height, SovSpecId::Genesis, l1_fee_rate);
+
+        // deploy selfdestruct contract
+        // send some money to the selfdestruct contract
+        // set some variable in the contract
+        let rlp_transactions = vec![
+            create_contract_message(&dev_signer, 0, SelfDestructorContract::default()),
+            send_money_to_contract_message(contract_addr, &dev_signer, 1, contract_balance as u128),
+            set_selfdestruct_arg_message(contract_addr, &dev_signer, 2, 123),
+        ];
+
+        evm.call(
+            CallMessage {
+                txs: rlp_transactions,
+            },
+            &context,
+            &mut working_set,
+        )
+        .unwrap();
+    }
+    evm.end_soft_confirmation_hook(&soft_confirmation_info, &mut working_set);
+    evm.finalize_hook(&[99u8; 32].into(), &mut working_set.accessory_state());
+
+    l2_height += 1;
+
+    let contract_info = evm
+        .accounts
+        .get(&contract_addr, &mut working_set)
+        .expect("contract address should exist");
+
+    // Test if we managed to send money to ocntract
```

Comment:
contract

Reply:
Literally saw it on my 4th time reading it 🔢